advanced_source/static_quantization_tutorial.rst

```diff
@@ -0,0 +1,675 @@
+            transforms.Compose([  
+                transforms.RandomResizedCrop(224),  
+                transforms.RandomHorizontalFlip(),  
+                transforms.ToTensor(),  
+                normalize,  
+            ])) 
+
+        dataset_test = torchvision.datasets.ImageFolder(  
+            valdir, 
+            transforms.Compose([  
+                transforms.Resize(256), 
+                transforms.CenterCrop(224), 
+                transforms.ToTensor(),  
+                normalize,  
+            ])) 
+
+        train_sampler = torch.utils.data.RandomSampler(dataset) 
+        test_sampler = torch.utils.data.SequentialSampler(dataset_test) 
+
+        data_loader = torch.utils.data.DataLoader(  
+            dataset, batch_size=train_batch_size, 
+            sampler=train_sampler)  
+
+        data_loader_test = torch.utils.data.DataLoader( 
+            dataset_test, batch_size=eval_batch_size, 
+            sampler=test_sampler) 
+
+        return data_loader, data_loader_test  
+
+
+Next, we'll load in the pre-trained MobileNetV2 model. We provide the URL to download the data from in ``torchvision``  
+`here <https://github.com/pytorch/vision/blob/master/torchvision/models/mobilenet.py#L9>`_. 
+
+.. code:: python
+
+    data_path = 'data/imagenet_1k'  
+    saved_model_dir = 'data/' 
+    float_model_file = 'mobilenet_pretrained_float.pth' 
+    scripted_float_model_file = 'mobilenet_quantization_scripted.pth' 
+    scripted_quantized_model_file = 'mobilenet_quantization_scripted_quantized.pth' 
+
+    train_batch_size = 30 
+    eval_batch_size = 30  
+
+    data_loader, data_loader_test = prepare_data_loaders(data_path) 
+    criterion = nn.CrossEntropyLoss() 
+    float_model = load_model(saved_model_dir + float_model_file).to('cpu')  
+ 
+    # Next, we'll "fuse modules"; this can both make the model faster by saving on memory access  
+    # while also improving numerical accuracy. While this can be used with any model, this is 
+    # especially common with quantized models.  
+
+    print('\n Inverted Residual Block: Before fusion \n\n', float_model.features[1].conv) 
+    float_model.eval()  
+
+    # Fuses modules 
+    float_model.fuse_model()  
+
+    # Note fusion of Conv+BN+Relu and Conv+Relu 
+    print('\n Inverted Residual Block: After fusion\n\n',float_model.features[1].conv)  
+
+  
+Finally to get a "baseline" accuracy, let's see the accuracy of our un-quantized model  
+with fused modules  
+
+.. code:: python
+
+    num_eval_batches = 10 
+
+    print("Size of baseline model") 
+    print_size_of_model(float_model)  
+
+    top1, top5 = evaluate(float_model, criterion, data_loader_test, neval_batches=num_eval_batches) 
+    print('Evaluation accuracy on %d images, %2.2f'%(num_eval_batches * eval_batch_size, top1.avg)) 
+    torch.jit.save(torch.jit.script(float_model), saved_model_dir + scripted_float_model_file)
+
+  
+We see 78% accuracy on 300 images, a solid baseline for ImageNet, 
```

Comment:
Let us change these numbers to full accuracy on imagenet eval data. We can also add that measuring the full accuracy takes about 5 minutes on a GPU.